### PR TITLE
fixes #16248: identifying searchType in validation function

### LINKED ISSUE #16248: Zoom Search is not performing input validation on INT columns
### Describe the bug
When we are using zoom search on a table validations are failing because of the fix id of table search form in validation function.

### To Reproduce

1. Select any table
2. Click on Search tabl
3. Click on Zoom Search subtab
4. Select any INT field in first / second row of first column
5. Enter invalid value in input field.

### Expected behavior
It should display appropriate message after successful validation just like in table serach form.

### Screenshots
Now
image

Expected After chaning id to zoom_search_form in validation function
image

pma_version: 5.0.3-dev
browser_name: CHROME
browser_version: 83.0.4103.116
user_os: Linux
server_software: Apache
php_version: 7.3.16

### PR DESCRIPTION
**Pull request by ankush-0x90**
Signed-off-by: Ankush Patil <aspraz2658@gmail.com>

### Description
Passing a searchType parameter in the validation function. Also needed to add event listener on comparison operators which are loaded after AJAX requests.

Still, There is little disturbance in UI when it shows the validation message.

Fixes #16248 

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

### CODE REVIEW

File: js/table/zoom_plot_jqplot.js [hunk 1]
```diff
@@ -129,6 +129,18 @@ AJAX.registerOnload('table/zoom_plot_jqplot.js', function () {
         searchedData = null;
     }
 
+    // adding event listener on select after AJAX request
+    var comparisonOperatorOnChange = function () {
+        var tableRows = $('#inputSection select.column-operator');
+        console.log(tableRows);
```
Comment (williamdes) [inline js/table/zoom_plot_jqplot.js:135]:
```suggestion
```

File: templates/table/search/selection_form.twig [hunk 2]
```diff
@@ -21,7 +21,8 @@
                 'keys' : keys,
                 'criteria_column_names': criteria_column_names,
                 'criteria_column_types': criteria_column_types
-            } only %}<table class="data">
+            } only %}
+            <table class="data">
```
Comment (williamdes) [inline templates/table/search/selection_form.twig:25]:
```suggestion
            } only %}<table class="data">
```

File: libraries/classes/Controllers/Table/SearchController.php [hunk 3]
```diff
@@ -979,7 +979,7 @@ public function getColumnProperties($search_index, $column_index)
             $type = 'INT';
         }
 
-        $html_attributes .= " onchange= 'return verifyAfterSearchFieldChange(" . $column_index . ")'";
+        $html_attributes .= " onchange= 'return verifyAfterSearchFieldChange(" . $column_index . ", \"" . $this->_searchType . "\")'";
```
Comment (williamdes) [inline libraries/classes/Controllers/Table/SearchController.php:982]: I would do the selector logic explicitly here :)
`#tbl_search_form`

Comment (asprazz) [reply libraries/classes/Controllers/Table/SearchController.php:982]: Ohk I will modify according to it :)

File: js/table/change.js [hunk 4]
```diff
@@ -151,16 +151,17 @@ function checkForCheckbox (multiEdit) {
...
     // validation for integer type
     if ($thisInput.data('type') === 'INT') {
         // Trim spaces if it's an integer
         $thisInput.val($thisInput.val().trim());
 
         var hasMultiple = $thisInput.prop('multiple');
+
         if (hasMultiple) {
-            $('#tbl_search_form').validate();
+            $('#' + searchFormId).validate();
```
Comment (williamdes) [inline js/table/change.js:164]:
```suggestion
            $(searchFormId).validate();
```

File: js/table/change.js [hunk 5]
```diff
@@ -170,7 +171,7 @@ function verifyAfterSearchFieldChange (index) {
             );
             validateMultipleIntField($thisInput, true);
         } else {
-            $('#tbl_search_form').validate();
+            $('#' + searchFormId).validate();
```
Comment (williamdes) [inline js/table/change.js:174]:
```suggestion
            $(searchFormId).validate();
```

File: libraries/classes/Controllers/Table/SearchController.php [hunk 6]
```diff
@@ -979,7 +979,9 @@ public function getColumnProperties($search_index, $column_index)
             $type = 'INT';
         }
 
-        $html_attributes .= " onchange= 'return verifyAfterSearchFieldChange(" . $column_index . ")'";
+        $search_form_id = $this->_searchType == 'zoom' ? 'zoom_search_form' : 'tbl_search_form';
```
Comment (williamdes) [inline libraries/classes/Controllers/Table/SearchController.php:982]:
```suggestion
        $search_form_id = $this->_searchType === 'zoom' ? '#zoom_search_form' : '#tbl_search_form';
```

**Comment by williamdes** [comment]
one last change ;)

**Comment by asprazz** [comment]
> one last change ;)

I have squashed commits again..   :)

**Review by williamdes — APPROVED** [review]

**Comment by williamdes** [comment]
Thank you @asprazz !

**Pull request merged**